### DISCUSSION TITLE
Windows compaitbility!

### DISCUSSION THREAD

**Discussion by TheBlewish** [TheBlewish/Automated-AI-Web-Researcher-Ollama#19] [Q&A]
Hi there, thanks very much hafeezhmha for attempting to make a windows compatible version of this program, I looked at your code and maybe I'm wrong I only had a quick glance, but I am concerned that your changes may make it fail to run on Linux?
Please let me know what you think, and even if it does I could, accept the pull request into a new branch called something like windows-support. but i'd love to get your input on this first!

Thanks mate!

**Comment by hafeezhmha** [comment]
@TheBlewish Hey James, Thanks for having a look at the code and yes, I think it might not work on linux because it uses a windows specific library. I have made lots of other changes, so the pull requests might change. I will make  updated pull requests again. 
Making a new branch for it would make sense imo.

Meanwhile, I'll see if i can work on making it compatible with both linux and windows.

**Reply by TheBlewish** [reply]
Thanks I appreciate that! And thanks for the hard work on this, I unfortunately don't have the technical capabilities (i.e. a windows computer) or the time (University) to do it myself so I really appreciate it!

Let me know when you have a working version even just for windows and I can make it a new branch and you can submit pull requests and I'll update it whenever you do so.

For now just focus on getting the windows one working when it does, let me know and I'll make the new branch for you, as the original already works on Linux so there's not much point in combining the two.

Especially since it would be an absolute pain to test given that you would only be able to test on windows for function then someone else or you would have to test on Linux the same code and any issues would be painstaking to then resolve like bugs and stuff. 

Anyways thanks mate! Let me know when you reckon you're windows version is somewhat functional and I'll go ahead and make that new branch!

**Reply by hafeezhmha** [reply]
Ah I see. It would be definitely be painstaking to resolve the bugs and stuff.

Alright. I shall let you know in sometime!

Thanks man! ^^

**Reply by TheBlewish** [reply]
No worries! Again if you get a windows version functioning just let me know!

**Reply by hafeezhmha** [reply]
Hey @TheBlewish You can create a new branch! I shall make a pull request. I have updated it and its working with the input handling. If some issue comes up, I'll patch them! Lemme know if you're okay with it.

**Reply by TheBlewish** [reply]
oh shit sorry I thought I said this in this discussion, I made a new branch! Which of course you already saw since you updated it.
Thanks for all your hard work and efforts! Now we have a viable windows version!

BTW There is a new issue related to I believe the windows version if you could look into it for me and see if it can be fixed that'd be AMAZING, thanks mate your a legend!

https://github.com/TheBlewish/Automated-AI-Web-Researcher-Ollama/issues/55

**Comment by MikeCrochet** [comment]
**[Koalageddon download](https://koalageddon.com/download/)** works on most Windows versions, but compatibility may vary depending on your system setup. Make sure to run it with admin rights and disable antivirus temporarily if you face issues.
